docs/products/kafka/howto/connect-with-go.rst

```diff
@@ -0,0 +1,278 @@
+Connect to Aiven for Apache Kafka® with Go
+=============================================
+
+These examples show how to connect to an Aiven for Apache Kafka® service using two different Kafka libraries in Go: `Sarama <https://github.com/Shopify/sarama>`_ and `Kafka-go <https://github.com/segmentio/kafka-go>`_.
+
+.. note::
+
+    The examples in this article provide two different options for authentication: SSL and SASL-SSL. For more information on these  authentication methods read :doc:`our article on Kafka authentication types <../concepts/auth-types>`.
+
+
+Pre-requisites
+---------------
+
+Go to the *Overview* page of your Aiven for Apache Kafka service.
+
+* If you are going to connect with SSL authentication:
+
+  * In the *Connection information* section:
+
+    #. If **Authentication Method** is shown, choose **Client Certificate**
+    #. Next to *Access Key*, click **Download** and save the ``service.key`` file.
+    #. Next to *Access Certificate*, click **Download** and save the ``service.cert`` file.
+    #. Next to *CA Certificate*, click **Download** and save the ``ca.pem`` file.
+
+* If you are going to connect using SASL authentication:
+
+  #. Follow the instructions at `Use SASL Authentication with Apache Kafka® <https://developer.aiven.io/docs/products/kafka/howto/kafka-sasl-auth.html>`_ to enable SASL.
+
+  #. In the *Connection Information* section
+
+     #. Select **SASL** as the **Authentication Method**
+     #. Next to *CA Certificate*, click **Download** and save the ``ca.pem`` file
+     #. Note the *Password* required for the SASL, we'll need it for authentication
+
+* Created the keystore ``client.keystore.p12`` and truststore ``client.truststore.jks`` by following  :doc:`our article on configuring Java SSL to access Kafka <../howto/keystore-truststore>`
+
+.. Warning::
+
+  In the below examples, we just pass the name of the keystore and truststore files, but in actual use, the full path should be used.
+
+Variables
+---------
+
+========================     =======================================================================================================
+Variable                     Description
+========================     =======================================================================================================
+``HOST``                     Host name for the connection
+``USER_NAME``                Name of the user for the connection
+``SSL_PORT``                 Port number to use for SSL
+``SASL_PORT``                Port number to use for SASL
+``SASL_PASSWORD``            Password required to connect using SASL
```

Comment:
As `USER_NAME` and `SASL_PASSWORD` are always used together, maybe change the ordering such that those two are next to each other? Also variable name could be `SASL_USERNAME` to clarity.

Reply:
Good idea! changed